### ISSUE TITLE
`@discardableResult` ignored when using `@MainActor` (Xcode 14.0.1)

### ISSUE THREAD

**Issue by 417-72KI** [apple/swift#61597]
**Describe the bug**
#60545 still alive in Xcode 14.0.1.

**Steps To Reproduce**

```swift
@MainActor
class Foo {
    @discardableResult
    func bar() -> Int {
        return 10
    }
}

@MainActor
@discardableResult
func baz() -> Bool {
    return true
}

@discardableResult
func qux() -> Double {
    return 0
}

func reproduceIssue() {
    Task {
        await Foo().bar()
        await baz()
        qux()
        return
    }
}

@MainActor
func reproduceIssue2() {
    Foo().bar()
    baz()
    qux()
}
```

**Expected behavior**
No warnings on each lines.

**Screenshots**
<img width="900" alt="image" src="https://user-images.githubusercontent.com/4150060/196141300-c1e6e1a5-ada4-4a60-b60a-697313a7eace.png">

**Environment (please fill out the following information)**
 - OS:  macOS 12.6
 - Xcode Version: **14.0.1 (14A400)**

**Comment by theblixguy** [comment]
The fix was merged a bit late into the 5.7 branch, so it didn't make the cut for Xcode 14 or 14.0.1, but I can verify it's fixed on Xcode 14.1 beta.

**Comment by 417-72KI** [comment]
I verified it fixed on Xcode 14.1 RC (14B47). Thx, @theblixguy !
